### ISSUE TITLE
Menu Item MediaFinder field does not load saved value

### ISSUE THREAD

**Issue by c-andrews** [rainlab/pages-plugin#256]
Adding the below added a "MenuImage" field to the menu item. It allows me to set an image and the value is saved into the menus YAML file in the themes Meta folder.

```php
public function boot()
	{
		Event::listen('backend.form.extendFields', function ($widget) {

			if (
				!$widget->getController() instanceof \RainLab\Pages\Controllers\Index ||
				!$widget->model instanceof \RainLab\Pages\Classes\MenuItem
			) {
				return;
			}

			$widget->addTabFields([
				'viewBag[menuImage]' => [
					'tab' => 'Media',
					'label' => 'Menu Image',
					'commentAbove' => 'Add an image to appear in the menu',
					'type' => 'Cms\FormWidgets\MediaFinder',
					'mode' => 'image',
					'span' => 'left'
				]
			]);
		});
	}
```
However when the menu item is reopened the value for the image set with the MediaFinder is not loaded so when any changes are re-applied the value saved is removed.

I would expect the value to be loaded by the MediaFinder so that the user can see what has previously been set and so previous settings are not removed.

Is there a way to resolve this issue?

**Comment by c-andrews** [comment]
Hi, any news on a possible solution for this issue?

**Comment by LukeTowers** [comment]
Have you tried setting `'mediafinder'` instead of `'Cms\FormWidgets\MediaFinder'`?

**Comment by c-andrews** [comment]
Yes I've tried both

**Comment by LukeTowers** [comment]
Hmm, well, canned response incoming:

The support demand for October is increasing, while the available time the founders have to give remains the same. These issues will be handled in time, however there are ways you can help:

- Submit a pull request that fixes this bug and wait for it to be reviewed
- [Post a bug bounty](https://www.bountysource.com/teams/october) for this specific issue so that more people can become involved with creating a fix
- [Make a financial contribution to the project](http://octobercms.com/fundraising) so we can allocate more time towards making improvements like these that benefit everyone

**Comment by jan-vince** [comment]
I have a same problem.

**No form widget** can be used in menu item - only standard field types (like text, textarea) works.

It looks like widgets cannot read a saved value from a menu item object. 
When media finder field type is changed from ````mediafinder```` to ````text````, saved image path is shown - so it is correctly saved.

I wasn't able to find a reason why form widgets doesn't work :( I will continue looking and testing but maybe someone has a solution and can save me some time ...

**Comment by jan-vince** [comment]
I was finally able to enable media finder support in Menu items by editing script ````loadProperties```` in
````plugins\rainlab\pages\formwidgets\menuitems\assets\js\menu-items-editor.js```` as @daftspunk  suggested in #185.

*I will add support for color picker as well and try to make PR ASAP.*

**Comment by c-andrews** [comment]
Brilliant, thank you for pointing me to that. I've now managed to get the images appearing by updating the `setPropertyOnElement` function inside `loadProperties`. This is what i've done:

```
var setPropertyOnElement = function($input, val) {

	if ($input.prop('type') == 'checkbox') {
		var checked = !(val == '0' || val == 'false' || val == 0 || val == undefined || val == null)
		checked ? $input.prop('checked', 'checked') : $input.removeAttr('checked')
	}
	else if ($input.prop('type') == 'radio') {
		$input.filter('[value="'+val+'"]').prop('checked', true)
	}
//--- ADDED START
	else if ($input.prop('type') == 'hidden') {

		var $parent = $input.parent();

		if ( $parent.data("control") == "mediafinder" )
		{
			$input.val(val)
			$input.change()

			if ( val &&  val.length > 1 )
			{
				var publicUrl = "/storage/app/media"+ val;

				$('[data-find-image]', $parent).attr('src', publicUrl)
				$('[data-find-file-name]', $parent).text( val.substring(1))

				$parent.addClass( "is-populated" );
			}
		}
	}
//--- ADDED END
	else {
		$input.val(val)
		$input.change()
	}
}
```

Is there a way to get the set public url path instead of hard coding it?

**Comment by jan-vince** [comment]
I have sent a PR #272 with support for mediafinder and colorpicker (as I use both in my projects).

You can look inside of the PR to see my changes - and feel free to send here your comments. I am not very good with JS.

*I am using OctoberCMS configured media path in data attribute added to ````_menuitems```` partial.*

**Comment by LukeTowers** [comment]
Fixed by #286
